src/utils.rs

```diff
@@ -0,0 +1,17 @@
+use crate::UrlParseError;
+use url::Url;
+
+pub fn check_uri(uri: &String) -> Result<Url, UrlParseError> {
+    let uri = Url::parse(uri.as_str());
+
+    let uri = match uri {
+        Ok(res) => res,
+        Err(err) => return Err(UrlParseError::Parser(err)),
+    };
+
+    if uri.scheme() != "https" {
+        return Err(UrlParseError::NotHttps);
+    }
```

Comment:
What's the reasoning behind mandating `https`? If Firebase has a problem with `http`, it should probably be reported from their side, not manually from the library.